searx/engines/chinaso.py

```diff
@@ -0,0 +1,143 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""ChinaSo: A search engine from ChinaSo."""
+
+from urllib.parse import urlencode
+from datetime import datetime
+
+from searx.exceptions import SearxEngineAPIException
+from searx.utils import html_to_text
+
+about = {
+    "website": "https://www.chinaso.com/",
+    "wikidata_id": "Q10846064",
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+paging = True
+time_range_support = True
+results_per_page = 10
+categories = []
+chinaso_category = 'news'
+"""ChinaSo supports news, videos, images search.
+
+- ``news``: search for news
+- ``videos``: search for videos
+- ``images``: search for images
+"""
+
+time_range_dict = {'day': '24h', 'week': '1w', 'month': '1m', 'year': '1y'}
+
+base_url = "https://www.chinaso.com"
+
+
+def request(query, params):
+    query_params = {"q": query}
+
+    if time_range_dict.get(params['time_range']):
+        query_params["stime"] = time_range_dict[params['time_range']]
+        query_params["etime"] = 'now'
+
+    category_config = {
+        'news': {'endpoint': '/v5/general/v1/web/search', 'params': {'pn': params["pageno"], 'ps': 10}},
+        'images': {
+            'endpoint': '/v5/general/v1/search/image',
+            'params': {'start_index': (params["pageno"] - 1) * 10, 'rn': 10},
+        },
+        'videos': {
+            'endpoint': '/v5/general/v1/search/video',
+            'params': {'start_index': (params["pageno"] - 1) * 10, 'rn': 10},
+        },
+    }
+
+    if chinaso_category not in category_config:
```

Comment:
The check here and the one in line 72 should be moved to the `init` function which runs when SearXNG is started:
```python
def init(_):
    if chinaso_category not in parsers:
        raise SearxEngineAPIException(f"Unsupported category: {chinaso_category}")
` ``